src/main/java/seedu/address/logic/commands/AddLessonCommand.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CLASS_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DAY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END_TIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NUMBER_OF_OCCURRENCES;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_TIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VENUE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.components.name.Name;
+import seedu.address.model.lesson.Lesson;
+import seedu.address.model.moduleclass.ModuleClass;
+
+/**
+ * Adds a lesson to the lesson manager.
+ */
+public class AddLessonCommand extends Command {
+
+    public static final String COMMAND_WORD = "add-lesson";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a lesson to the lesson manager. "
+            + "Parameters: "
+            + PREFIX_CLASS_INDEX + "CLASS_INDEX (must be a positive integer) "
+            + PREFIX_DAY + "DAY "
+            + PREFIX_START_TIME + "START_TIME "
+            + PREFIX_END_TIME + "END_TIME "
+            + PREFIX_VENUE + "VENUE "
+            + PREFIX_NUMBER_OF_OCCURRENCES + "NO_OF_OCCURRENCES\n"
```

Comment:
Would it be better specify the range of values accepted here as well?
```suggestion
            + PREFIX_NUMBER_OF_OCCURRENCES + "NO_OF_OCCURRENCES (must be a positive integer between 1 and 52\n"
```